File: Data Structures & Algorithms/anagram-groups/submission-5.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> anag;
        for(auto str : strs){
            unordered_map<int, int> mp;
            for(auto c : str){
                mp[c-'a']++;
            }
            string key;
            for(int i = 0; i < 26; i++){
                key += ','+mp[i];
            }
            anag[key].push_back(str);
        }
        for(auto i : anag){
            res.push_back(i.second);
        }
        return res;
    }
};

File: Data Structures & Algorithms/is-subsequence/submission-5.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        lt, ls = len(t)-1, len(s)-1
        while lt >= 0:
            if ls < 0:
                return True
            if s[ls] == t[lt]:
                ls -=1
                lt -=1
            else:
                lt -=1
            
            if ls < 0:
                return True
        return False
